src/rtapi/rtapi_compat.c

```diff
@@ -246,11 +243,6 @@ flavor_ptr flavor_byid(int flavor_id)
 flavor_ptr default_flavor(void)
 {
     char *fname = getenv("FLAVOR");
```

Comment:
I am strangely reluctant to mess around with it at present :stuck_out_tongue_closed_eyes: 

Need to look at the calls `flavor` makes and the way kernel type assumptions arise.
I had to change it once before with the machinekit-NG work, because it kept getting it wrong.

Now that we have removed hard rtai and xenomai kernels, it really should be quite simple to determine the running kernel.  

Just grepping kern.log on an x86 will get you all you need
```
Linux version 4.6.5-rt10 (root@INTEL-i7) (gcc version 4.9.2 (Debian 4.9.2-10) ) #1 SMP PREEMPT RT Sun Aug 21 12:07:48 BST 2016
Command line: BOOT_IMAGE=/boot/vmlinuz-4.6.5-rt10 root=UUID=b25ba191-a27c-414d-9a42-10ebcaea2ad4 ro quiet
```
Not sure about arm systems with uboot though